src/etc/inc/plugins.inc.d/unbound.inc

```diff
@@ -282,9 +283,17 @@ EOD;
             foreach ($dnsservers as $dnsserver) {
                 if (strpos($dnsserver, '127.') === 0 || $dnsserver == '::1') {
                     $forward_local = "do-not-query-localhost: no\n";
+                } else {
+                    /* Generate a custom resolv.conf file for use by unbound-anchor.
+                     * These servers all use port 53 so exclude localhost from being queried for bootstrapping
+                     * in our custom resolv.conf file as Unbound doesn't exist yet.
+                     */
+                    $resolv .= "nameserver $dnsserver\n";
                 }
                 $forward_conf .= "\tforward-addr: $dnsserver\n";
             }
+
+            file_put_contents("/var/unbound/resolv.conf.root", $resolv, LOCK_EX);
```

Comment:
are we always writing the file (empty in worst case)? to make sure it is reset if settings changed

Reply:
good point. it should be removed or emptied out if forwarding is disabled.